### PR TITLE
Login as user

### PR DESCRIPTION
**Pull request by lyyder**
Adds two endpoints to the server that handle authorization code grant type login requests. Enables operators to log in as a user.

Takes the new `isAnonymous` attribute returned by `sdk.authInfo` to resolve if a user is logged in.

A banner notifying a user in case their access token has limited write access will be added in a distinct PR.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,7 +51,7 @@
     "redux": "^4.0.1",
     "redux-thunk": "^2.3.0",
     "seedrandom": "^3.0.3",
-    "sharetribe-flex-sdk": "^1.8.0",
+    "sharetribe-flex-sdk": "1.9.0",
```
Comment (Gnito) [inline package.json]: Was there a reason to remove caret?

File: server/index.js [hunk 2]
```diff
@@ -132,6 +133,9 @@ if (!dev) {
   }
 }
 
+// Server-side routes that do not render the application
+app.use('/api', apiRouter);
```
Comment (Gnito) [inline server/index.js]: Could you also add a note to here:
https://github.com/sharetribe/ftw-daily/blob/master/src/routeConfiguration.js#L48

Just to ensure that if there's a clash of existing code in some of the customizations the first step is to look from routeConfiguration that what's wrong there. Some code comment would give a hint that people need to take a look to server/index.js too.

File: package.json (hunk 1 above)
Comment (lyyder) [reply package.json]: No, that's just a mistake. 👍

File: server/apiRouter.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+response_type=code&\
+client_id=${CLIENT_ID}&\
+redirect_uri=${loginAsRedirectUri}&\
+user_id=${userId}&\
+state=${state}&\
+code_challenge=${codeChallenge}&\
+code_challenge_method=S256`;
+
+  res.cookie(stateKey, state);
+  res.cookie(codeVerifierKey, codeVerifier);
```
Comment (Gnito) [inline server/apiRouter.js]: I assume that SDK clears these tokens automatically when logout happens. I didn't test though.

File: server/apiRouter.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+ * replacing unaccepted characters.
+ */
+const urlifyBase64 = base64Str =>
+  base64Str
+    .replace(/\+/g, '-')
+    .replace(/\//g, '_')
+    .replace(/=/g, '');
+
+// Creates a 302 response that redirects to start an OAuth2 Authorization code
+// login flow.
```
Comment (Gnito) [inline server/apiRouter.js]: Could you elaborate more here? I.e. what are the steps/redirections in this process.

Comment (lyyder) [reply server/apiRouter.js]: I'll try.

File: server/apiRouter.js (hunk 3 above)
Comment (bladealslayer) [reply server/apiRouter.js]: Also, these two cookies should be flagged as secure, if possible (e.g. when HTTPS is enabled). Shouldn't they also have some (short) expiration time? Were there nowadays any other recommended security-related flags or settings for cookies?

Comment (lyyder) [reply server/apiRouter.js]: Added a 30 second ttl to the cookies and retract them after they've been read.

**Pull request merged**